### DISCUSSION TITLE
AWS costs per Hubs Cloud instance (history)

### DISCUSSION THREAD

**Discussion by NigelRidderhof** [mozilla/hubs#4949] [Hubs Cloud Questions] [answered]
Hi! 

We have 5 Hubs Cloud instances running in AWS at the moment. Recently, we wanted to chart how much we pay per month for each instance, since they all belong to a different client of ours and costs are combined within the billing. 

These are the forum discussions I read: 
https://forums.aws.amazon.com/thread.jspa?threadID=169892 
https://stackoverflow.com/questions/63719507/how-can-i-find-out-cost-per-instance-per-month-on-aws
https://stackoverflow.com/questions/68548256/how-to-get-ec2-instance-costs-via-aws-billing

There seem to be 2 ways to go about getting insight into the costs per instance / client.
Option 1. Creating sub-accounts in AWS for each client underneath a parent account, so that you get seperate bills. 
Option 2. Activating Cost Allocation Tags within AWS, so that you can group by Tag > Name within the Cost Explorer. 

One of the main problems is that neither of these options give us information about the previous costs / the history.
Option 1 would I guess mean we would need to redo our entire stack set-ups for all 5 clients, which is really not preferable to do long after initialising and using the stacks / platforms. Since I think we would lose the specefic url's, Spoke scene's, etc. 

And within option 2, costs are still combined together. 
As you can see in the screenshot below, the seperation of the costs is only working from the moment I activated the (User-defined) Cost Allocation Tag > Name  at the end of November. And costs are still being combined whithin "No Tagkey: Name" and "app-db", as you can see within the red marked area's.

![image](https://user-images.githubusercontent.com/43579707/146176260-46046b81-aded-47be-aba3-ae1e2f426230.png)

I've also tried activating more (User-defined) Cost Allocation Tags (client, stack-name, cloudformation:stack-name), so that maybe all the costs would be seperated and not clumped together. But even though it then says they are activated, most still have not come through after multiple weeks. (And not to mention the fact that all this activating things for this cost purpuse might hove broken one or more of our stacks, since we were not able to build anymore - as you can read in a discussion I placed earlier.) 

Does anyone know a better way to get insight into the (previous) costs per instance? Or know how to build upon my discussed attempt with the Cost Allocation Tags / Cost Explorer?

I think this could definitely be of value for more Hubs Cloud creators. Cheers!

**Comment by pattersonbl2** [comment, marked as answer]
Knowing how AWS does billing this may be something that can't be done until you use something like AWS organization.  You can create a separate account for each client under one organization.  I have only known aws to lump all resource cost into single groups by resources.

**Reply by NigelRidderhof** [reply]
Allright, at least good to know! Thankyou!
